internal/kubectl.go

```diff
@@ -177,6 +179,64 @@ func (c Kubectl) Get(resource, namespace string, w io.Writer) error {
 	return printer.PrintObj(obj, w)
 }
 
+// GetElastic retrieves the Elastic K8s objects of type resourceName in namespace
+// using a field selector generated from the label selector and marshals them into the writer w.
+func (c Kubectl) GetElastic(resourceName, namespace string, filter filters.Filter, w io.Writer) error {
+	var (
+		err error
+		r   *resource.Result
+	)
+
+	if filter.LabelSelector() != "" {
+		typ := filter.Type()
+		name := filter.Name()
+		if typ != resourceName {
+			return nil
+		}
+		r, err = c.getResourcesWithFieldSelector(resourceName, namespace, fmt.Sprintf("metadata.name=%s", name))
+		if err != nil {
+			return err
+		}
+	} else {
+		r, err = c.getResources(resourceName, namespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	printer, err := printers.NewTypeSetter(scheme.Scheme).WrapToPrinter(&printers.JSONPrinter{}, nil)
+	if err != nil {
+		return err
+	}
+
+	obj, err := r.Object()
+	if err != nil {
+		return err
+	}
+
+	return printer.PrintObj(obj, w)
+}
+
+func extractTypeName(selectors string) (string, string, error) {
```

Comment:
only used in test

Reply:
Remnant of the previous implementation.  Thank you.  Removed